codec/unknownproto/unknown_fields_test.go

```diff
@@ -446,49 +446,9 @@ func TestRejectUnknownFieldsNested(t *testing.T) {
 			recv: new(testdata.TestVersion1),
 			wantErr: &errMismatchedWireType{
 				Type:         "*testdata.TestVersion3",
-				TagNum:       1,
-				GotWireType:  2,
-				WantWireType: 0,
-			},
-		},
-		{
-			name: "From nested proto message, message index 0",
```

Comment:
@odeke-em I'm pretty sure this test case shouldn't error, imo TestVersion3LoneNesting's bytes can be unmarshalled into TestVersion4LoneNesting

Reply:
Okay in that case then perhaps don't delete the entire cases, please modify them accordingly, and explain in a comment. Thank you.